trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/signing/SigningController.java

```diff
@@ -90,37 +90,31 @@ public String signRequest(
 
     public SigningMetadata validateAndParseAuthorization(Request request, SigningServiceType signingServiceType)
     {
-        if (!request.parsedAuthorization().isValid()) {
+        if (!request.requestAuthorization().isValid()) {
             throw new WebApplicationException(Response.Status.UNAUTHORIZED);
         }
 
-        return credentialsController.withCredentials(request.parsedAuthorization().accessKey(), request.session(), credentials -> {
+        return credentialsController.withCredentials(request.requestAuthorization().accessKey(), request.session(), credentials -> {
             SigningMetadata metadata = new SigningMetadata(signingServiceType, credentials, Optional.empty());
             return isValidAuthorization(metadata, request, Credentials::emulated);
         }).orElseThrow(() -> new WebApplicationException(Response.Status.UNAUTHORIZED));
     }
 
     public Optional<InputStream> inputStreamForContent(
```

Comment:
It looks like `inputStreamForContent` is no longer needed in `SigningController` now. We can use the few lines of code here directly instead.

Reply:
Sounds good: will also move `AwsChunkedInputStream` to the `rest` package - it makes sense and it also lets us keep it package-private after implementing this suggestion